### PR TITLE
Half of second task

### PR DESCRIPTION
**Pull request by Spectra456**

### CODE REVIEW

File: ImageProcessing/transformations.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+def Log(file):
+    image, draw, width, height, pixel = properities(file)
+    array = np.asarray(image)
+
+    for i in range(width):
+        for j in range(height):
+            a = pixel[i, j][0]
+            b = pixel[i, j][1]
+            c = pixel[i, j][2]
+            S = a * 0.2126 + b * 0.7152 + c * 0.0722
```
Comment (denklewer) [inline ImageProcessing/transformations.py]: what do these magic digits mean?   
Why can't you use log transformation like this? : x[i][j] = C * Math.Log10(x[i][j] + 1);

File: ImageProcessing/transformations.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            b = pixel[i, j][1] + factor
+            c = pixel[i, j][2] + factor
+
+            if (a < 0):
+                a = 0
+            if (b < 0):
+                b = 0
+            if (c < 0):
+                c = 0
+            if (a > 255):
```
Comment (denklewer) [inline ImageProcessing/transformations.py]: what does it mean?  
Saying about rescaling image I meant something like this:
C# : https://pastebin.com/zYSd0eWa
Python : https://pastebin.com/pMQn252P

**Review by denklewer — CHANGES_REQUESTED** [review]

File: ImageProcessing/transformations.py (hunk 2 above)
Comment (Spectra456) [reply ImageProcessing/transformations.py]: It's not rescaling image, it's Gamma correction of image.
For rescaling i use opencv in file zoom.py

File: ImageProcessing/transformations.py (hunk 1 above)
Comment (Spectra456) [reply ImageProcessing/transformations.py]: It's formula for converting RGB images to grayscale images.
I use log transformation in line 69

**Pull request merged**